src/app/qgisapp.cpp

```diff
@@ -10837,6 +10838,41 @@ void QgisApp::projectCrsChanged()
   }
 }
 
+void QgisApp::projectTemporalRangeChanged()
+{
+  QMap<QString, QgsMapLayer *> layers = QgsProject::instance()->mapLayers();
+  QgsMapLayer *currentLayer = nullptr;
+
+  for ( QMap<QString, QgsMapLayer *>::const_iterator it = layers.constBegin(); it != layers.constEnd(); ++it )
+  {
+    currentLayer = it.value();
+
+    if ( currentLayer->dataProvider() )
+    {
+      QgsDataSourceUri uri;
+      uri.setEncodedUri( currentLayer->dataProvider()->dataSourceUri() );
+
+      if ( uri.hasParam( QStringLiteral( "temporalSource" ) ) &&
```

Comment:
Where can I find some documentation about "temporalSource" ? From where is that set? (I'm wondering how to support that in PG rasters)

Reply:
I have used "temporalSource" to store where the WMS-T layers get the static temporal range from, can either be "project" or "provider". This is set at first from the wms provider when loading the wms-t layer for the first time and from the raster layer properties when changing static temporal range options.